client/src/screens/group.screen.js

```diff
@@ -94,41 +178,131 @@ class Group extends Component {
     ),
   };
 
-  keyExtractor = item => item.id;
+  onRefresh() {
+    this.props.refetch();
+  }
 
-  renderItem = ({ item }) => <Member member={item} />
+  joinOrLeaveGroup(group, memberAlready) {
+    const { joinGroup, leaveGroup, refetch } = this.props;
+    if (!memberAlready) {
+      joinGroup({ groupId: group.id }).then(() => {
+      // dont need to do anything, but we might in the future
```

Comment:
I understand why you put this comment here but I think it's probably best to leave it out - we can add the `.then()` in the future if we need it.